scripts/tui-pty-smoke.py

```
@@ -0,0 +1,434 @@
+def run(binary: str, args: list[str], rows: int, columns: int, mouse: bool, secret_probe: bool = False, home: str | None = None, term: str | None = None, command: list[str] | None = None) -> None:
+    home_context = tempfile.TemporaryDirectory(prefix="veto-tui-pty-") if home is None else None
+    selected_home = home if home is not None else home_context.name
+    try:
+        pid, master = pty.fork()
+        if pid == 0:
+            env = os.environ.copy()
+            env.update({"HOME": selected_home, "NO_COLOR": "1"})
+            if term is not None:
+                env["TERM"] = term
+            argv = [binary, "tui", *args] if command is None else [binary, *command]
+            os.execve(binary, argv, env)
```

Ruff (0.16.3)

[error] Starting a process without a shell

(S606)